src/dev/lupio/lupio_pic.cc

```diff
@@ -74,37 +80,45 @@ LupioPIC::clear(int src_id)
 void
 LupioPIC::lupioPicUpdateIRQ()
 {
-    if (nThread > 1 ) {
-        panic("This device currently does not have SMP support\n");
-    }
+    int cpu;
+
+    for (cpu = 0; cpu < nThread; cpu++) {
+        auto tc = system->threads[cpu];
 
-    auto tc = system->threads[0];
-    if (pending & mask) {
-        tc->getCpuPtr()->postInterrupt(tc->threadId(), intType, 0);
-    } else {
-        tc->getCpuPtr()->clearInterrupt(tc->threadId(), intType, 0);
+        if (enable[cpu] & mask[cpu] & pending) {
+            tc->getCpuPtr()->postInterrupt(tc->threadId(), intType, 0);
+        } else {
+            tc->getCpuPtr()->clearInterrupt(tc->threadId(), intType, 0);
+        }
     }
 }
 
 uint64_t
 LupioPIC::lupioPicRead(uint8_t addr)
 {
     uint32_t r = 0;
-    
-    switch (addr >> 2) {
+    int cpu, reg;
```

Comment:
Put these declarations on the same line as the initialization. We can use C++-17+, no need to restrict ourselves to C-98